app/src/main/java/com/abhijitvalluri/android/fitnotifications/AppSettingsActivity.java

```diff
@@ -116,7 +118,8 @@ public void onClick(View v) {
                                                                            mStartTimeMinute,
                                                                            mStopTimeHour,
                                                                            mStopTimeMinute,
-                                                                           R.string.start_time_heading,
+                                                                           stopTimeFormatted,
```

Comment:
You are right. I didn't realize the formatted value will be "captured" and will remain the same even after the time was changed.

I believe it's friendlier to show the user which time we expect his input to be before/after.
So I moved the formatting to the fragment's `onCreateDialog`. It will now use exact same hour/minute as in the sanity check.
`is24hFormat` parameter was redundant too. So I removed it as well.

And I should be doing more thorough tests indeed.